jupyter_client/client.py

```diff
@@ -181,9 +181,13 @@ def hb_channel(self):
 
     def is_alive(self):
         """Is the kernel process still running?"""
+        from .manager import KernelManager
+        if isinstance(self.parent, KernelManager):
+            # We were created by a KernelManager, we can ask them:
```

Comment:
``# Kernel client was created by a KernelManager, we can ask the parent KernelManager: ``